### PR TITLE
Secrets/Auth Mount Path Whitespace Warning 

### LINKED ISSUE #8268: TrimSpace secret engine path
**Is your feature request related to a problem? Please describe.**
I created a secret engine, and was unable to access it afterwards through most tools (CLI, Ansible lookup module). After debugging it for an hour, I came across the space in the beginning of my secret path. I accidentally entered a space before the path.

**Describe the solution you'd like**
When I create a new Secrets Engine, it would be awesome that the path will be trimmed - just in case. I don't see any reason why a space would be useful in this place.

**Describe alternatives you've considered**
I haven't really considered any other solutions. Maybe trimming in the Vault UI or showing a notification ("You have a space in front of the path, are you sure?") would also suffice. 

**Explain any additional use-cases**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by zofskeez**
This PR builds on the alert banner that was added to the secrets path input to warn of values containing whitespace and adds it to the secrets and auth mount path inputs. Since `FormFieldGroups` are used I added an attribute option `whitespaceWarning` and a `has-whitespace` helper to be used in the `FormField` template for the standard text input. This could be expanded to other form field types if need be.

image

image

Closes #8268

### CODE REVIEW

File: ui/app/templates/components/secret-create-or-update.hbs [hunk 1]
```diff
@@ -34,12 +34,11 @@
           <code>/</code>
         </p>
       {{/if}}
-      {{#if this.pathWhiteSpaceWarning}}
+      {{#if (has-whitespace (get @modelForData @modelForData.pathAttr))}}
         <div class="has-top-margin-m">
           <AlertBanner
             @type="warning"
             @message="Your secret path contains whitespace. If this is desired, you'll need to encode it with %20 in API calls."
-            @marginTop={{true}}
```
Comment (zofskeez) [inline ui/app/templates/components/secret-create-or-update.hbs:42]: Double checked and this isn't an arg on `AlertBanner`

**Review by hashishaw — APPROVED** [review]
Pretty straightforward! I do wonder if there might be more warnings in the future, it might be better to abstract the warning logic into the validations decorator rather than a one-off helper and component-check. It feels a little strange to define a warning in a different place than where we defined errors. But I'll leave that up to you!

File: ui/app/models/secret-engine.js [hunk 2]
```diff
@@ -24,7 +24,10 @@ const validations = {
 @withModelValidations(validations)
 @withExpandedAttributes()
 export default class SecretEngineModel extends Model {
-  @attr('string') path;
+  @attr('string', {
+    whitespaceWarning: true,
```
[the remark this answers is not in the document]
Comment (zofskeez) [reply ui/app/models/secret-engine.js:28]: We could go that route but we would need to expand the model validations to allow for warnings that do not block saving. That would also require a secondary event on key change to be added.

**Comment by zofskeez** [comment]
> it might be better to abstract the warning logic into the validations decorator rather than a one-off helper and component-check

Great suggestion! I explored this further and it will definitely be a more scalable solution for other warning cases that might arise in the future. I'm going to close this PR in favor of the suggested approach. Thanks for your eyes on this and feedback!

**Pull request closed without merging**